src/device.rs

```diff
@@ -123,6 +129,77 @@ impl<SPI, NCS, E> Device for SpiDevice<SPI, NCS>
     }
 }
 
+impl<SPI, NCS, E> AK8963 for SpiDevice<SPI, NCS>
+    where SPI: spi::Write<u8, Error = E> + spi::Transfer<u8, Error = E>,
+          NCS: OutputPin
+{
+    type Error = E;
+
+    fn init<D: DelayMs<u8>>(&mut self,
+                            delay: &mut D)
+                            -> Result<(), Self::Error> {
+        // Isolate the auxiliary master I2C bus (AUX_CL, AUX_DA)
+        // disable the slave I2C bus, make serial interface SPI only
+        // reset the master I2C bus
+        Device::write(self, Register::USER_CTRL, 0x32)?;
+        delay.delay_ms(10);
+        Ok(())
+    }
+
+    fn finalize<D: DelayMs<u8>>(&mut self, delay: &mut D) -> Result<(), Self::Error> {
+        // set aux I2C frequency to 400 KHz (should be configurable?)
+        Device::write(self, Register::I2C_MST_CTRL, 0x0d)?;
+
+        delay.delay_ms(10);
+
+        // configure sampling of magnetometer
+        Device::write(self, Register::I2C_SLV0_ADDR, ak8963::I2C_ADDRESS | ak8963::R)?;
+        Device::write(self, Register::I2C_SLV0_REG, ak8963::Register::XOUT_L.addr())?;
+        Device::write(self, Register::I2C_SLV0_CTRL, 0x87)?;
+
+        delay.delay_ms(10);
+        Ok(())
+    }
+
+    fn read(&mut self, reg: ak8963::Register) -> Result<u8, Self::Error> {
+        Device::write(self,
+                      Register::I2C_SLV4_ADDR,
+                      ak8963::I2C_ADDRESS | ak8963::R)?;
+        Device::write(self, Register::I2C_SLV4_REG, reg.addr())?;
+
+        // start transfer
+        Device::write(self, Register::I2C_SLV4_CTRL, 0x80)?;
+
+        // wait until transfer is over
+        while Device::read(self, Register::I2C_MST_STATUS)? & (1 << 6) == 0 {}
```

Comment:
Btw, I guess we need to add `max_reads` or smth to prevent apps from hanging indefinetely, when ak8963 damaged or dosn't respond for some reason. Out of scope of this PR, of course.